Fix documentation regarding .NET Framework support

pellared opened the pull request:
## Why

Related PR and commit:
- https://github.com/signalfx/signalfx-dotnet-tracing/pull/238
- https://github.com/signalfx/signalfx-dotnet-tracing/pull/238/commits/8e1dd7dd29aec78506482709096f246b77c80d41

## What

We support .NET Framework 4.6.1+

## Tests

Automated tests are running on  .NET Framework 4.6.1

Kielek commented on docs/README.md:
```diff
@@ -35,7 +35,7 @@ and [`ActivitySource`](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/
 ## Requirements
 
 - .NET Core 3.1, .NET 5.0 and higher on Windows and Linux (except Alpine Linux)
-- .NET Framework 4.6.2 and higher on Windows
```
I suppose this typo could stay with us, especially due to "Support for .NET Framework 4.5.2, 4.6, and 4.6.1 will end on April 26, 2022"

from https://docs.microsoft.com/en-us/lifecycle/faq/dotnet-framework

Kielek replied:
Accepting, I do not have any objection to merge it for now.

pellared commented:
I have also fixed other minor typos and grammar errors.
Updated https://github.com/signalfx/signalfx-dotnet-tracing/releases/tag/v0.2.0 description as well.
